src/main/scala/tech/sourced/gemini/ReportSparkApp.scala

```diff
@@ -2,20 +2,58 @@ package tech.sourced.gemini
 
 import com.datastax.driver.core.Cluster
 
+case class ReportAppConfig(host: String = Gemini.defaultCassandraHost,
+                           port: Int = Gemini.defaultCassandraPort,
+                           mode: String = ReportSparkApp.defaultMode
+                          )
+
 object ReportSparkApp extends App {
-  def printUsage(): Unit = {
-    println("Usage: ./report [--use-group-by|--condensed]")
-    println("")
-    println("Finds duplicate files among hashed repositories." +
-      "\n\tIt uses as many queries as distinct files are stored in the database"
-    )
-    println("  --use-group-by" +
-      "\n\tIt uses as many queries as unique duplicate files are found, plus one. (only for Apache Cassandra database)"
-    )
-    println("  --condensed displays only the duplicate blob_hash and the number of occurrences." +
-      "\n\tIt uses only one query to find the duplicates. (only for Apache Cassandra database)"
-    )
-    System.exit(2)
+  val defaultMode = ""
+  val groupByMode = "use-group-by"
+  val condensedMode = "condensed"
```

Comment:
What do you think if we keep just 2 flags:
 - `--use-group-by`
 - `--no-group-by` _(default)_

Reply:
the `--condensed`  one is missing, in case we want to keep it:
It reports (using Cassandra) the duplicate `blob_hash` found in the db with only one (GROUP BY) query
![image](https://user-images.githubusercontent.com/2437584/34212637-37f689b8-e59d-11e7-9cac-5336ce74b8aa.png)

Reply:
@dpordomingo I see! Do you think that for a user, this only would make sense together with `--use-group-by` then ?

If so, do you think we could keep it i.e in the debug output of `--use-group-by`, or can you seen a value of this command for the end user? 

In my perspective, I would say it makes sense to avoid supporting anything cassandra-specific at this point, so I would refrain from having an extra flag.

Reply:
@bzz An alternative (maybe clearer) could be:
```bash
report --enable-group-by=[true|false] --condensed=[true|false]
* default modes are false for both
```

If we choose that one, it is already implemented the following combinations:
- `report --enable-group-by=false --condensed=false`<br />already done as: `report`
- `report --enable-group-by=true --condensed=false`<br />already done as: `report --use-group-by`
- `report --enable-group-by=true --condensed=true`<br />already done as: `report --condensed`

If so, it should be implemented:
- `report --enable-group-by=false --condensed=true`

If you agree, I can do it taking this PR as a dependency :dancer:

Reply:
If we do that way, we can drop the support for `--enable-group-by` at any moment, and keep the `--condensed` behavior if it's still useful.

Reply:
I see, that makes sense.

But to be hones, I find `condensed` quite confusing. I would avoid changes like `--enable-group-by=[true|false]` in this PR -  it sounds complex for both, users to use and developer to maintain.

At this point, I would strongly prefer to have only 1 extra switch for `./hash` - to enable "Cassandra" behavior, which is off by default.  

There are other tools that provide plenty of CLI options for finding similar files, and I see having very simple CLI UX as a value proposition, that differentiates Gemini.

Reply:
In my opinion, we need to think about it a bit more. At least currently we have 2 different printers. If we remove one we need to decide which one to keep.
I think it's out of the scope of current PR.

Master branch has 2 options:
- `--use-group-by`
- `--condensed`
and you can't use both at the same time:
https://github.com/src-d/gemini/pull/11/files#diff-aa8cc8ee5a5bc3498c0d159a611380abR40

This PR does exactly the same, but instead of 2 options it has 1 key-value option:
- mode=group-by
- mode=condensed

I've created separate issue for this: https://github.com/src-d/gemini/issues/23